### PR TITLE
Keys Stats API

### LINKED ISSUE #564: [TASK] Add keys section hooks
## Overview 
Graphql queries and wrapping hooks for keys section

## Checklist
- [x] Listing all keys with pagination and page info
- [x] Provide data for details view
- [x] Provide data for the current active key and the next key

### LINKED ISSUE #567: [TASK] Fetch the chain sync metadata
## Overview 
Fetch metadata for the currently connected chain to SubQuery, most required data is the latest synced block

### PR DESCRIPTION
**Pull request by AhmedKorim**
## Summary of changes
- scaffolding statistics API integration with the dApp
- Wrapped the key UI backend

### Reference issue to close (if applicable)
- Closes #564 
- Closes #567 

-----
### Code Checklist 

- [x] Tested
- [x] Documented

### CODE REVIEW

File: packages/page-statistics/src/AuthoritiesData.tsx [hunk 1]
```diff
@@ -17,22 +16,15 @@ const AuthoritiesDataWrapper = styled.div`
...
-  if (error) {
-    return <div>Something went wrong</div>;
-  }
-
+  const data = useKeys({
+    offset: 0,
+    perPage: 30,
+  });
+  const activeKeys = useActiveKeys();
+  console.log(activeKeys);
```
Comment (dutterbutter) [inline packages/page-statistics/src/AuthoritiesData.tsx:24]: Can we remove this console log please.

File: packages/page-statistics/src/generated/graphql.tsx [hunk 2]
```diff
@@ -2213,6 +2223,8 @@ export enum BlocksOrderBy {
   ProposalItemsStddevPopulationVotesDesc = 'PROPOSAL_ITEMS_STDDEV_POPULATION_VOTES_DESC',
   ProposalItemsStddevSampleBlockIdAsc = 'PROPOSAL_ITEMS_STDDEV_SAMPLE_BLOCK_ID_ASC',
   ProposalItemsStddevSampleBlockIdDesc = 'PROPOSAL_ITEMS_STDDEV_SAMPLE_BLOCK_ID_DESC',
+  ProposalItemsStddevSampleBlockNumberAsc = 'PROPOSAL_ITEMS_STDDEV_SAMPLE_BLOCK_NUMBER_ASC',
+  ProposalItemsStddevSampleBlockNumberDesc = 'PROPOSAL_ITEMS_STDDEV_SAMPLE_BLOCK_NUMBER_DESC',
```
Comment (dutterbutter) [inline packages/page-statistics/src/generated/graphql.tsx:2227]: What are the sample items?

File: packages/page-statistics/src/gql/proposals.graphql [hunk 3]
```diff
@@ -1,5 +1,5 @@
-query Proposals {
-  proposalItems {
+query Proposals($PerPage: Int!, $Offset: Int!) {
+  proposalItems(first: $PerPage, offset: $Offset) {
```
Comment (dutterbutter) [inline packages/page-statistics/src/gql/proposals.graphql:2]: Can we add relevant documentation comments for each of these queries (e.g. expected response and usage example). Unless they are documented elsewhere.

File: packages/page-statistics/src/provider/hooks/types.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
+/**
+ * A Type that wrappers an API call value for a list view
```
Comment (dutterbutter) [inline packages/page-statistics/src/provider/hooks/types.ts:2]:
```suggestion
 * A Type that wraps an API call value for a list view
```

File: packages/page-statistics/src/provider/hooks/types.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+export type Page<T> = {
+  items: T[];
+  pageInfo: {
+    count: number;
+    hasNext: boolean;
+    hasPrevious: boolean;
+  };
+};
+/**
+ * A Type that wrappers an API call value
```
Comment (dutterbutter) [inline packages/page-statistics/src/provider/hooks/types.ts:16]:
```suggestion
 * A Type that wraps an API call value
```

File: packages/page-statistics/src/provider/hooks/useKeys.ts [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+export interface PublicKeyListView extends PublicKeyContent {
+  height: string;
+  keyGenAuthorities: string[];
+  keyGenThreshold: string;
+  signatureThreshold: string;
+  session: string;
+}
+/**
+ * Public key progress history item
+ * @param at - The ate when the history item took place
```
Comment (dutterbutter) [inline packages/page-statistics/src/provider/hooks/useKeys.ts:55]:
```suggestion
 * @param at - The date when the history item took place
```

File: packages/page-statistics/src/AuthoritiesData.tsx (hunk 1 above)
Comment (dutterbutter) [inline packages/page-statistics/src/AuthoritiesData.tsx:24]: Can we remove this console log

**Review by dutterbutter — CHANGES_REQUESTED** [review]
Great job on this @AhmedKorim !!

**Review by AtelyPham — COMMENTED** [review]
I still think we have too many generated files, can we remove these generated files, then update the `README` for how to generate those files to run and develop the dApp. It would help reduce the repo size and reduce the effort to review the PR each time those files have changed. @AhmedKorim

File: packages/page-statistics/src/generated/graphql.tsx (hunk 2 above)
Comment (AhmedKorim) [reply packages/page-statistics/src/generated/graphql.tsx:2227]: The is from the code in Subql side will check that

**Comment by AhmedKorim** [comment]
@AtelyPham  in the meantime we don't have much to do about the generated files.Not sure you should review them, I will update the readme for the code stuff

**Review by dutterbutter — APPROVED** [review]

File: packages/page-statistics/README.md [hunk 7]
```diff
@@ -1 +1,9 @@
 @webb-dapp/page-mixer
+
+
+## Setup
+
+1. Run the graphql server for SubQuery
```
Comment (dutterbutter) [inline packages/page-statistics/README.md:6]: Add command to run `graphql server for SubQuery`

**Pull request merged**